### ISSUE TITLE
current sources complain with "tdlib version 1.7.9 is required"

### ISSUE THREAD

**Issue by itt533** [ars3niy/tdlib-purple#169]
i downloaded the current sources of tdlib-purple, and at cmake command prompt i get:
======================================
CMake Error at CMakeLists.txt:11 (message):
  tdlib version 1.7.9 is required

  (version found: 1.8.4)
======================================

my command is :
cd tdlib-purple-master
mkdir -p build
cd build
cmake \
    -DTd_DIR=/usr/lib/cmake/Td .. \
    -DTgvoip_LIBRARIES="tgvoip;opus"

**Comment by BenWiederhake** [comment]
Well, if tdlib version 1.7.9 is required, then why are you trying to build with version 1.8.4 instead?

**Comment by itt533** [comment]
yes it looks like i'm not serious, but the fact that it is requiring one exact version surprised me and also aur.archlinux.org is currently using v. 1.8.0. They are actually using the patch from the pull 154 to patch tdlib-purple v. 0.7.9.rxxx.

**Comment by BenWiederhake** [comment]
I'm very sorry to hear that aur.archlinux.org is using an unsupported combination of versions, but I cannot change that. You might want to use a supported combination, for example:

| tdlib-purple | td | works? |
| --- | --- | --- |
| 0.8.1 or `master` | 1.7.9 | Yes, like the build script says |
| https://github.com/ars3niy/tdlib-purple/pull/159 (or #154 if you insist) | 1.8.0 | Yes, that's what I'm using |
| ??? | 1.8.4 | No, at least not to my knowledge |

Of course, you're very welcome to update tdlib-purple to deal with td 1.8.4, I believe this would make many people happy.

EDIT: I typo'd the td version number, it's now fixed.

**Comment by itt533** [comment]
ok i tried and it worked, with #159 it's better as it downloads some of the history. But I'm just discovering that despite the effort of telegram to offer a fully open API to allow us making our own clients there's a long way to go with this plugin to be descent.

**Comment by BenWiederhake** [comment]
Like I said, you're very welcome to update tdlib-purple to deal with the newest td, I believe this would make many people happy. The only thing missing is one person who actually implements it.

**Comment by itt533** [comment]
i feel fine about that but is it worth it? I'd prefer to do it from scratch since the current version is not doing a lot. How long time the current version of pidgin is going to live?

**Comment by BenWiederhake** [comment]
So what is your question here? If you don't wanna use it, then don't use it.

**Comment by itt533** [comment]
i'm serious, is the current version of pidgin going to live long? I read in the telegram channel of tblib-purple that it lacks flexibility to implement all the features that telegram offers. The work of developing this plugin seems very big.

**Comment by BenWiederhake** [comment]
I'm a pidgin user, and this is the repository for a plugin particularly for pidgin, which I still use and sometimes make a few edits to keep the lights on. If that doesn't answer your question, then I don't know what will.

**Comment by asctime** [comment]
Wow this sucks. I hope that proper packaging and build documentation can be restored to this project. bleh...

**Comment by BenWiederhake** [comment]
You're welcome to improve the situation instead of complaining.

**Comment by asctime** [comment]
At the risk of helping someone who can't be helped, I'll clarify:
1) If your project requires _very_ specific dependencies then it should be on the front page of your project in **BOLD** type - it shouldn't be something "discovered" half way into a build.
2) Working links to hard to find package versions that WILL compile should be provided.
3) If your releases are necessarily married to specific dependency versions then it would probably be a good idea to even name them that way.
4) Proper forks should be offered for upstream compatibility versions (1.8.0 fork etc) not just dig through your patches and hope you did it right.
One more thing. Honestly, any package that changes it's public API *every* single minor release probably isn't worth working on to begin with..

**Comment by heeplr** [comment]
I think the main problem here is that there doesn't seem to be an official 1.7.9 release from upstream. All I could find was the [github tags](https://github.com/tdlib/td/tags).
Where could I get the 1.7.9 source?

**Comment by BenWiederhake** [comment]
You could just [use this script / repo to do everything in a single command](https://github.com/BenWiederhake/tdlib-purple/blob/master/build_and_install.sh), or if you want to do it the hard way: commit 1d1bc07eded7d3ee7df7c567e008bbf926cba081.
